bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2112,6 +2112,15 @@ void ReplicaImp::onMessage<ReplicaStatusMsg>(ReplicaStatusMsg *msg) {
             sendAndIncrementMetric(prePrepareMsg, msgSenderId, metric_sent_preprepare_msg_due_to_status_);
           }
         }
+        // help from Inactive Window
+        if (mainLog->isPressentInHistory(msg->getLastExecutedSeqNum() + 1)) {
+          for (SeqNum i = msg->getLastExecutedSeqNum() + 1; i < lastStableSeqNum + 1; i++) {
+            PrePrepareMsg *prePrepareMsg = mainLog->getFromHistory(i).getSelfPrePrepareMsg();
+            if (prePrepareMsg != nullptr && !msg->isPrePrepareInActiveWindow(i)) {
```

Comment:
!msg->isPrePrepareInActiveWindow(i) should probably be an assert since we're getting the messages from the history.

Reply:
This check is in order not to re-transmit unnecessarily. The Replica that sends Status Exchange request will state for which SeqNo-s it already has PrePrepares, which will be in its Active Window. For the Replica that is helping in this case the SeqNo will be in the Inactive Window.